### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by hopeforkoami**
In this branch, I:
 - Remove `current_user `method in `ApplicationController`
 - Install and setup devise.
   * Can register a new user.
   * User logs in with a combination of email and password.
   * Hashed password should be stored in the database.
   * Ask for confirmation of email.
   * Can reset password.
 - Modify the views in devise for registration and login to match the wireframes from the Sneak Peek and your styling.
 
Thanks

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit( "Log in", class:"button") %>
+  </div>
+<% end %>
```
Comment (omar-muhamad) [inline app/views/devise/sessions/new.html.erb]: - [ ] [Optional][Highly recommended] Please add a link that directs to the sign-up page, So when the dev server opens for the first time the user can go to the sign-up page from this link.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
```
Comment (omar-muhamad) [inline app/views/devise/registrations/new.html.erb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#:~:text=Can%20reset%20password.-,Modify%20the%20views%20in%20devise%20for%20registration%20and%20login%20to%20match%20the%20wireframes%20from%20the%20Sneak%20Peek%20and%20your%20styling.,-Need%20a%20big), please add styling to the sign_up page as you did with the log_in page.

- [ ] When I try to add a new user, It requires a username but I can't see an input field for the username on the sign_up page, please fix that.
[Hint] See this [post](https://stackoverflow.com/questions/42572124/adding-custom-parameters-to-devise-registration-unpermitted-parameters) may help.

https://user-images.githubusercontent.com/68796232/227748468-e50d7844-a2b5-4bfc-a24a-64932fa3067e.mp4

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @hopeforkoami,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShadyShawkat — APPROVED** [review]
## Hi @hopeforkoami, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — COMMENTED** [review]
Hi @hopeforkoami,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### This is a collaborative project, although it was approved you still need to redo this project from scratch with your new coding partner. If you have no coding partner then kindly provide proof like screenshots of your communication with student success.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**